File: app/src/main/java/com/example/testtanaw/util/MessageAdapter.java
package com.example.testtanaw.util;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.testtanaw.R;
import com.example.testtanaw.models.Message;
import java.util.List;

public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.MessageViewHolder> {
    private final List<Message> messages;

    public MessageAdapter(List<Message> messages) {
        this.messages = messages;
    }

    public static class MessageViewHolder extends RecyclerView.ViewHolder {
        public final TextView subject;
        public final TextView preview;
        public final TextView timestamp;

        public MessageViewHolder(View view) {
            super(view);
            subject = view.findViewById(R.id.message_subject);
            preview = view.findViewById(R.id.message_preview);
            timestamp = view.findViewById(R.id.message_timestamp);
        }
    }

    @NonNull
    @Override
    public MessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_message, parent, false);
        return new MessageViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MessageViewHolder holder, int position) {
        Message message = messages.get(position);
        holder.subject.setText(message.getSubject());
        holder.preview.setText(message.getPreview());
        holder.timestamp.setText(message.getTimestamp());

        // Highlight unread messages
        int textColor = message.isRead() ? R.color.black : R.color.black;
        holder.subject.setTextColor(holder.itemView.getContext().getColor(textColor));
    }

    @Override
    public int getItemCount() {
        return messages.size();
    }
}

File: app/src/main/java/com/example/testtanaw/MessageAdapter.kt
package com.example.testtanaw

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.testtanaw.R

class MessageAdapter(
    private val messages: List<Message>
) : RecyclerView.Adapter<MessageAdapter.MessageViewHolder>() {

    class MessageViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val subject: TextView = view.findViewById(R.id.message_subject)
        val preview: TextView = view.findViewById(R.id.message_preview)
        val timestamp: TextView = view.findViewById(R.id.message_timestamp)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MessageViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_message, parent, false)
        return MessageViewHolder(view)
    }

    override fun onBindViewHolder(holder: MessageViewHolder, position: Int) {
        val message = messages[position]
        holder.subject.text = message.subject
        holder.preview.text = message.preview
        holder.timestamp.text = message.timestamp

        // Highlight unread messages
        val textColor = if (message.isRead) R.color.black else R.color.black
        holder.subject.setTextColor(holder.itemView.context.getColor(textColor))
    }

    override fun getItemCount(): Int = messages.size
}
